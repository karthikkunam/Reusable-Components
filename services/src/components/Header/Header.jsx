import React from 'react';
import PropTypes from 'prop-types';
import './styles/Header.scss';
import HamBurgerMenu from './headerComponents/HamBurgerMenu';
import Logo from './headerComponents/Logo';
import StoreId from './headerComponents/StoreId';
import EmployeeName from './headerComponents/EmployeeName';
import ClockInClockOut from './headerComponents/ClockInClockOut';
import PrintPage from './headerComponents/PrintPage';
import AlertIcon from './headerComponents/AlertIcon';
import OtherLinks from './headerComponents/OtherLinks';
import { logIn } from '../../constants';
import HamBurgerSubMenu from './headerComponents/HamBurgerSubMenu';

class Header extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      leftNavOpen: false,
      showSubMenu: false,
      storeDetails: [{ storeId: '', fullName: '' }],
      subMenuItems: [],
    };
  }

  componentDidMount() {
    if (this.props.headerType !== 'loggedIn') {
      sessionStorage.removeItem('storeInfo');
    } else {
      this.setStoreDetails();
    }
  }

  componentDidUpdate(prevProps) {
    if (this.props.storeDetails !== prevProps.storeDetails) {
      if (this.props.headerType !== 'loggedIn') {
        sessionStorage.removeItem('storeInfo');
      } else {
        this.setStoreDetails();
      }
    }
  }

  setStoreDetails = () => {
    let getSessionStoreInfo = null;
    if (this.props.storeDetails) {
      sessionStorage.removeItem('storeInfo');
      this.setState({ storeDetails: this.props.storeDetails });
    } else {
      if (window.sessionStorage.getItem('storeInfo')) {
        getSessionStoreInfo = JSON.parse(window.sessionStorage.getItem('storeInfo'));
      }
    }
    if (!getSessionStoreInfo) {
      return;
    }
    this.setState({ storeDetails: getSessionStoreInfo });
  };

  leftNavToggle = () => {
    const { leftNavItems } = this.props;
    if (leftNavItems && leftNavItems.length > 0) {
      this.setState({
        leftNavOpen: !this.state.leftNavOpen,
      });
    }
  };

  handleLeftMenu = subMenus => {
    const { leftNavOpen } = this.state;
    if (subMenus && leftNavOpen) {
      this.setState({ subMenuItems: subMenus, showSubMenu: true });
    }
  };

  render() {
    const { leftNavOpen, storeDetails, subMenuItems, showSubMenu } = this.state;
    const {
      headerType,
      logoClassName,
      hideHamburgerMenu,
      hideClockIn,
      hidePrinter,
      hideNotifications,
      hideOtherLink,
      leftNavItems,
      onMenuChange,
      otherLink,
      defaultLeftNav,
      defaultSubNav,
    } = this.props;

    return (
      <>
        <nav className="navbar navbar-expand-md navbar-dark bg-dark fixed-top logged-in">
          <div className="container">
            {headerType === 'loggedIn' && !hideHamburgerMenu && (
              <button
                onClick={this.leftNavToggle}
                className={`navbar-toggler navbar-toggler-left${!leftNavOpen ? ' collapsed' : ''}`}
                type="button"
                data-toggle="collapse"
                data-target="#navbarNavDropdown1"
                aria-controls="navbarNavDropdown1"
                aria-expanded="false"
                aria-label="Toggle navigation">
                <span className="icon icon-hamburger"></span>
              </button>
            )}
            <Logo logoClassName={logoClassName} />
            {headerType === 'loggedIn' && (
              <div id="navbarNavDropdown" className="collapse navbar-collapse header-info">
                <StoreId storeID={storeDetails[0].storeId} />
                <EmployeeName employeeName={storeDetails[0].fullName} />
                {!hideClockIn && <ClockInClockOut />}
                <ul className="navbar-nav image-container">
                  {!hidePrinter && (
                    <li className="nav-item">
                      <PrintPage />
                    </li>
                  )}
                  {!hideNotifications && (
                    <li className="nav-item">
                      <AlertIcon />
                    </li>
                  )}
                </ul>
                {!hideOtherLink && <OtherLinks link={otherLink} />}
              </div>
            )}
            {headerType === 'loggedOut' && (
              <button type="submit" className="btn btn-sm btn-outline-primary">
                {logIn}
              </button>
            )}
          </div>
        </nav>
        {/* SIDE NAV MENU LEFT */}
        <div id="navbarNavDropdown1" className={`sidenav left ${leftNavOpen && 'show'}`}>
          <div className="container level-0">
            <HamBurgerMenu
              leftNavItems={leftNavItems}
              handleLeftMenu={this.handleLeftMenu}
              defaultLeftNav={defaultLeftNav}
              expanded={leftNavOpen}
            />
          </div>
          <div id="navbarNavDropdown11" className={`container level-1 ${showSubMenu && leftNavOpen && 'show'}`}>
            <HamBurgerSubMenu subMenuItems={subMenuItems} onMenuChange={onMenuChange} defaultSubNav={defaultSubNav} />
          </div>
        </div>
      </>
    );
  }
}

Header.propTypes = {
  logoClassName: PropTypes.string,
  headerType: PropTypes.string,
  hideHamburgerMenu: PropTypes.bool,
  hideClockIn: PropTypes.bool,
  hidePrinter: PropTypes.bool,
  hideNotifications: PropTypes.bool,
  hideOtherLink: PropTypes.bool,
  leftNavItems: PropTypes.arrayOf(
    PropTypes.shape({
      label: PropTypes.string,
      iconClass: PropTypes.string,
      url: PropTypes.string,
    }),
  ),
  defaultLeftNav: PropTypes.number,
  defaultSubNav: PropTypes.string,
  storeDetails: PropTypes.objectOf({
    storeId: PropTypes.string,
    fullName: PropTypes.string,
  }),
  onMenuChange: PropTypes.func,
  otherLink: PropTypes.arrayOf(
    PropTypes.shape({
      label: PropTypes.string,
      url: PropTypes.string,
    }),
  ),
};

Header.defaultProps = {
  logoClassName: 'navbar-brand-logo header-logo img-fluid',
  hideHamburgerMenu: false,
  hideClockIn: false,
  hidePrinter: false,
  hideNotifications: false,
  hideOtherLink: false,
  otherLink: {
    label: '7HUB',
    url: 'https://fam.7-eleven.com/idp/startSSO.ping?PartnerSpId=ca.ondemand.saml.20.post.7ELEVENINC-Prod',
  },
};

export default Header;
