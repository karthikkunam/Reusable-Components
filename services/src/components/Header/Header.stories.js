import React from 'react';
import Header from '.';
import { action } from '@storybook/addon-actions';
import ModalButton from '../Modal/ModalButton.jsx';
export default {
  title: 'Header',
  component: Header,
};
import SessionManger from '../Session/SessionManager';

export const headerWithoutLogin = () => <Header />;
export const headerWithLogin = () => <Header headerType="loggedOut" />;

const leftNavItems = [
  {
    leftMenu: 'Home',
    subMenus: [
      {
        header: 'Breadcrumb',
        subMenuList: [
          {
            title: 'Basic Breadcrumb',
            url: '/?path=/story/breadcrumb--basic-bread-crumb',
          },
        ],
      },
      {
        header: 'Button',
        subMenuList: [
          {
            title: 'Button variant',
            url: '/?path=/story/button--variant',
          },
          {
            title: 'Button variant',
            url: '/?path=/story/button--outline',
          },
          {
            title: 'Disabled Button',
            url: '/?path=/story/button--disable',
          },
        ],
      },
    ]
  },
  {
    leftMenu: 'Product Assortment',
    subMenus: [
      {
        header: 'Exception Message',
        subMenuList: [
          {
            title: 'Success exception message',
            url: '/?path=/story/exceptionmessage--success-message',
          },
        ],
      },
    ]
  },
  {
    leftMenu: 'ordering',
  },
  {
    leftMenu: 'Inventory Mangement',
  },
  {
    leftMenu: 'Cash Management',
  },
  {
    leftMenu: 'Payroll',
  },
  {
    leftMenu: 'Gasoline Management',
  },
  {
    leftMenu: 'System Configuration',
  },
];

const leftNavWithTwoItems = [
  {
    leftMenu: 'home',
    subMenus: [
      {
        header: 'Header 1',
        subMenuList: [
          {
            title: 'page 1 Title',
            url: 'page1',
          },
          {
            title: 'page 2 Title',
            url: 'page2',
          },
          {
            title: 'page 3 Title',
            url: 'page3',
          },
        ],
      },
      {
        header: 'Header 2',
        subMenuList: [
          {
            title: 'page 4 Title',
            url: 'page4',
          },
          {
            title: 'page 5 Title',
            url: 'page5',
          },
        ],
      },
    ],
  },
  {
    leftMenu: 'ordering',
    subMenus: [
      {
        header: 'Header 1',
        subMenuList: [
          {
            title: 'page 6 Title',
            url: 'page6',
          },
          {
            title: 'page 7 Title',
            url: 'page7',
          },
          {
            title: 'page 8 Title',
            url: 'page8',
          },
        ],
      },
      {
        header: 'Header 2',
        subMenuList: [
          {
            title: 'page 9 Title',
            url: 'page9',
          },
          {
            title: 'page 10 Title',
            url: 'page10',
          },
        ],
      },
    ],
  },
  {
    label: 'ordering',
    iconClass: 'icon-ordering',
    url: 'ordering',
  },
];

const otherLink = {
  label: '7HUB',
  url: 'https://fam.7-eleven.com/idp/startSSO.ping?PartnerSpId=ca.ondemand.saml.20.post.7ELEVENINC-Prod',
};

const tokenDetails = {
  refreshTokenUrl: '/7boss/order/auth/refresh/token',
  payLoad: {},
  authorization:
    'eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJmaXJzdE5hbWUiOiJzZXJ2aWNlQWNjb3VudDMiLCJsYXN0TmFtZSI6Imp1bXBjbG91ZFNlcnZpY2UiLCJ1c2VybmFtZSI6InNhanVtcGNsb3VkM0A3MTEuY29tIiwiZW1haWwiOiJzYWp1bXBjbG91ZDNANzExLmNvbSIsImRldmljZVR5cGUiOiJyZW1vdGVBY2Nlc3MiLCJpc1N1cHBvcnRVc2VyIjpmYWxzZSwiYWxsb3dlZFN0b3JlcyI6W3sic3RvcmVJZCI6IjE4NDczIiwidGltZVpvbmUiOiJDU1QiLCJpc0FjdGl2ZSI6dHJ1ZSwiYWRkcmVzcyI6eyJzdHJlZXRBZGRyZXNzIjoiMTQwNCBXIFBSQVRUIEFWRSIsImNpdHkiOiJDSElDQUdPIiwic3RhdGUiOiJJTCIsInppcCI6IjYwNjI2NDIxNiIsImNvdW50cnkiOiJVUyIsInBob25lIjo3NzM5NzMxNjA3fSwidXNlclR5cGUiOiJTdG9yZSBNYW5hZ2VyIn0seyJzdG9yZUlkIjoiMzcwNjciLCJ0aW1lWm9uZSI6IlBTVCIsImlzQWN0aXZlIjp0cnVlLCJhZGRyZXNzIjp7InN0cmVldEFkZHJlc3MiOiIzOTU2NiBMT1MgQUxBTU9TIiwiY2l0eSI6Ik1VUlJJRVRBIiwic3RhdGUiOiJDQSIsInppcCI6IjkyNTYzIiwiY291bnRyeSI6IlVTIiwicGhvbmUiOjk1MTMwNDc4OTF9LCJ1c2VyVHlwZSI6IlN0b3JlIE1hbmFnZXIifSx7InN0b3JlSWQiOiIzNTQwOCIsInRpbWVab25lIjoiQ1NUIiwiaXNBY3RpdmUiOnRydWUsImFkZHJlc3MiOnsic3RyZWV0QWRkcmVzcyI6IjIwMjEgUyBNQUlOIFNUIiwiY2l0eSI6IktFTExFUiIsInN0YXRlIjoiVFgiLCJ6aXAiOiI3NjI0OCIsImNvdW50cnkiOiJVUyIsInBob25lIjo4MTc3NDEyODAwfSwidXNlclR5cGUiOiJTdG9yZSBNYW5hZ2VyIn0seyJzdG9yZUlkIjoiMzYzNjQiLCJ0aW1lWm9uZSI6IkNTVCIsImlzQWN0aXZlIjp0cnVlLCJhZGRyZXNzIjp7InN0cmVldEFkZHJlc3MiOiI3NzAgVyBMWU5ET04gQiBKT0hOU09OIiwiY2l0eSI6IklSVklORyIsInN0YXRlIjoiVFgiLCJ6aXAiOiI3NTA2MyIsImNvdW50cnkiOiJVUyIsInBob25lIjoyMTQ0OTYwMjE2fSwidXNlclR5cGUiOiJTdG9yZSBNYW5hZ2VyIn0seyJzdG9yZUlkIjoiMzUzNzgiLCJ0aW1lWm9uZSI6IkNTVCIsImlzQWN0aXZlIjp0cnVlLCJhZGRyZXNzIjp7InN0cmVldEFkZHJlc3MiOiI4Mzc1IE4gQkVMVExJTkUgUkQiLCJjaXR5IjoiSVJWSU5HIiwic3RhdGUiOiJUWCIsInppcCI6Ijc1MDYzNjAwNSIsImNvdW50cnkiOiJVUyIsInBob25lIjo5NzI2MjEyMjQ1fSwidXNlclR5cGUiOiJTdG9yZSBNYW5hZ2VyIn1dLCJpYXQiOjE1ODQ2NDUyMzAsImV4cCI6MTU4NDY0NjEzMH0.ca_1LnKHQPuLvV3ioRg-5PVKk1Tc4w5xBVda9Lp8yEGMAjYgSC7t4f4FBHMslZuhjxrp7MGileNouPmZMC4jHx8hM5l-s02VMiN0bFdmHYiBzhsZBVAanOiK-IisWjKVd2U7mLKoUJ3dap3w02nMgewubEuAYQoe5A4Bp1WiZYbW5h0IfzZXWLgFHpLGLjzC5SXAnrNriVsiFxls450fDFCpSQo1ocPBy9SGM7qRhb8rO644Yly1QGo8AhNFshI6SYqMLo0pqT8f-aNJ9no7FEZSgH7Dt2dEiEov73mKFb1zClOBTBZZ2EVpfB4UnS71gvYDS9icMEzpro2WobzUMAut_IFcZ9xm-JATf_enpELaqLw0tPJHweQTLgOM7ueuuzKSYYgabnZz3-8258nRvk8_N2GjpjG6Q3G7jewEvIP5OpP-MEOPnJn6vpWDwGpCi1xDlvaWx2xFKXVzADPTnDvJj3sqcduG46VMod0Xe3Y1mvE7-MiagIwacUqgNj9GclnP1jKuJ5cdQlOit4MXEIO1T4Fas-AUMh-kHu9B_otmSvGX0KXLw4s3Fnnj0Bn7yiZKNgr6CJfb3cJJ_Hv22HelejwH3RVNPJT05RYq9MOvZg4smiy3CQrdE5XDqA6MvhPUuJmD1-SVtulZEhhVzlJKHVR-ukSNl6uJnlO_kDA',
};

const storeDetails = [
  {
    storeId: '54768',
    fullName: '7BOSS User',
  },
];

export const headerAfterLoggedIn = () => (
  <>
    <SessionManger />
    <Header
      headerType="loggedIn"
      leftNavItems={leftNavItems}
      onMenuChange={action('Clicked')}
      otherLink={otherLink}
      logoClassName="navbar-brand-logo header-logo img-fluid"
      storeDetails={storeDetails}
    />
  </>
);
export const headerWithDefaultLeftNav = () => (
  <>
    <SessionManger />
    <Header
      headerType="loggedIn"
      leftNavItems={leftNavItems}
      onMenuChange={action('Clicked')}
      otherLink={otherLink}
      logoClassName="navbar-brand-logo header-logo img-fluid"
      storeDetails={storeDetails}
      defaultLeftNav={0}
    />
  </>
);
export const headerWithTwoHamburgerItems = () => (
  <>
    <SessionManger />
    <Header
      headerType="loggedIn"
      storeDetails={storeDetails}
      leftNavItems={leftNavWithTwoItems}
      onMenuChange={action('Clicked')}
      otherLink={otherLink}
    />
  </>
);
export const HeaderWithoutHamburger = () => (
  <>
    <SessionManger />
    <Header
      headerType="loggedIn"
      leftNavItems={leftNavItems}
      onMenuChange={action('Clicked')}
      otherLink={otherLink}
      logoClassName="navbar-brand-logo header-logo img-fluid"
      hideHamburgerMenu={true}
      storeDetails={storeDetails}
    />
  </>
);
export const HeaderWithoutLeftNav = () => (
  <>
    <SessionManger />
    <Header
      headerType="loggedIn"
      onNavigationChange={action('Clicked')}
      otherLink={otherLink}
      logoClassName= 'navbar-brand-logo header-logo img-fluid'
      storeDetails={storeDetails}
    />
  </>
);

export const headerWithoutStoreIdAndName = () => (
  <>
    <SessionManger tokenDetails={tokenDetails} />
    <Header
      headerType="loggedIn"
      leftNavItems={leftNavWithTwoItems}
      onMenuChange={action('Clicked')}
      otherLink={otherLink}
    />
  </>
);

export const headerWithoutSessionManager = () => (
  <>
    <Header
      headerType="loggedIn"
      leftNavItems={leftNavItems}
      onMenuChange={action('Clicked')}
      otherLink={otherLink}
      logoClassName="navbar-brand-logo header-logo img-fluid"
      hideHamburgerMenu={true}
      storeDetails={storeDetails}
    />
  </>
);
export const headerWithAllHideOptions = () => (
  <Header
    headerType="loggedIn"
    leftNavItems={leftNavItems}
    onMenuChange={action('Clicked')}
    otherLink={otherLink}
    hideHamburgerMenu={true}
    hideClockIn={true}
    hidePrinter={true}
    hideNotifications={true}
    hideOtherLink={true}
    storeDetails={storeDetails}
  />
);

export const headerWithContent = () => (
  <>
    <Header headerType="loggedIn" otherLink={otherLink} storeDetails={storeDetails} />
    <ModalButton
      primaryButton={{ buttonLabel: 'Save Changes', className: 'btn-success' }}
      secondaryButton={{ buttonLabel: 'Close', className: 'btn-secondary' }}
      onModalClose={action('Modal closed')}
    />
  </>
);
