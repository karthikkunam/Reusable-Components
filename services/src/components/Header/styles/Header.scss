.page {
    &-header {
        margin-top: 15px;
        &-title {
            display: inline-block;
        }
        &-subtitle {
            margin: 0;
            padding: 28px 0 25px;
            font-size: 14px;
            float: right;
        }
    }
}
