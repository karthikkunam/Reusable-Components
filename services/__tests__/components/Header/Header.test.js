import React from 'react';
import Header from '../../../src/components/Header/Header.jsx';
import { shallow } from 'enzyme';

const defaultProps = {
  headerType: '',
  logoClassName: '',
  hideHamburgerMenu: false,
  hideClockIn: false,
  hidePrinter: false,
  hideNotifications: false,
  hideOtherLink: false,
  leftNavOpen: false,
  leftNavItems: [
    {
      label: 'home',
      iconClass: 'icon-home',
      url: 'home',
    },
    {
      label: 'inventory management',
      iconClass: 'icon-promotions',
      url: 'inventory-management',
    },
  ],
  onNavigationChange: jest.fn(),
  otherLink: {
    label: '7HUB',
    url: 'https://fam.7-eleven.com/idp/startSSO.ping?PartnerSpId=ca.ondemand.saml.20.post.7ELEVENINC-Prod',
  },
};
describe('Header Component', () => {
  it('renders defaultProps correctly', () => {
    const component = shallow(<Header {...defaultProps} />);
    expect(component).toMatchSnapshot();
  });
  it('should render storeDetails from props', () => {
    const props = {
      storeDetails: [
        {
          storeId: '54768',
          fullName: '7BOSS',
        },
      ],
    };
    const component = shallow(<Header {...defaultProps} {...props} />);
    expect(component).toMatchSnapshot();
  });
  it('should render loggedOut header', () => {
    const props = {
      headerType: 'loggedOut',
    };
    const component = shallow(<Header {...defaultProps} {...props} />);
    expect(component).toMatchSnapshot();
  });
  it('should open leftNavToggle onClick', () => {
    const props = {
      headerType: 'loggedOut',
      leftNavOpen: false
    };
    const component = shallow(<Header {...defaultProps} {...props} />);
    expect(component).toMatchSnapshot();
    component.find('button').at(0).simulate('click', {
      leftNavToggle: () =>{},
    });
  });
  it('render component without HamburgerMenu', () =>{
    const props = {
      hideHamburgerMenu: true,
    };
    const component = shallow(<Header {...defaultProps} {...props} />);
    expect(component).toMatchSnapshot();
  });
  it('render component without ClockIn', () =>{
    const props = {
      hideClockIn: true,
    };
    const component = shallow(<Header {...defaultProps} {...props} />);
    expect(component).toMatchSnapshot();
  });
  it('render component without printer', () =>{
    const props = {
      hidePrinter: true,
    };
    const component = shallow(<Header {...defaultProps} {...props} />);
    expect(component).toMatchSnapshot();
  });
  it('render component without Notifications', () =>{
    const props = {
      hideNotifications: true,
    };
    const component = shallow(<Header {...defaultProps} {...props} />);
    expect(component).toMatchSnapshot();
  });
  it('render component without otherLinks', () =>{
    const props = {
      hideOtherLink: true,
    };
    const component = shallow(<Header {...defaultProps} {...props} />);
    expect(component).toMatchSnapshot();
  });
});
