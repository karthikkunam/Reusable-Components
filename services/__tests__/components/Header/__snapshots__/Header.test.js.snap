// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component render component without ClockIn 1`] = `
<Fragment>
  <nav
    className="navbar navbar-expand-md navbar-dark bg-dark fixed-top logged-in"
  >
    <div
      className="container"
    >
      <Logo
        logoClassName=""
      />
    </div>
  </nav>
  <div
    className="sidenav left false"
    id="navbarNavDropdown1"
  >
    <div
      className="container level-0"
    >
      <HamBurgerMenu
        defaultLeftNav={null}
        handleLeftMenu={[Function]}
        leftNavItems={
          Array [
            Object {
              "iconClass": "icon-home",
              "label": "home",
              "url": "home",
            },
            Object {
              "iconClass": "icon-promotions",
              "label": "inventory management",
              "url": "inventory-management",
            },
          ]
        }
      />
    </div>
    <div
      className="container level-1 false"
    >
      <HamBurgerSubMenu
        defaultSubNav={null}
        subMenuItems={Array []}
      />
    </div>
  </div>
</Fragment>
`;

exports[`Header Component render component without HamburgerMenu 1`] = `
<Fragment>
  <nav
    className="navbar navbar-expand-md navbar-dark bg-dark fixed-top logged-in"
  >
    <div
      className="container"
    >
      <Logo
        logoClassName=""
      />
    </div>
  </nav>
  <div
    className="sidenav left false"
    id="navbarNavDropdown1"
  >
    <div
      className="container level-0"
    >
      <HamBurgerMenu
        defaultLeftNav={null}
        handleLeftMenu={[Function]}
        leftNavItems={
          Array [
            Object {
              "iconClass": "icon-home",
              "label": "home",
              "url": "home",
            },
            Object {
              "iconClass": "icon-promotions",
              "label": "inventory management",
              "url": "inventory-management",
            },
          ]
        }
      />
    </div>
    <div
      className="container level-1 false"
    >
      <HamBurgerSubMenu
        defaultSubNav={null}
        subMenuItems={Array []}
      />
    </div>
  </div>
</Fragment>
`;

exports[`Header Component render component without Notifications 1`] = `
<Fragment>
  <nav
    className="navbar navbar-expand-md navbar-dark bg-dark fixed-top logged-in"
  >
    <div
      className="container"
    >
      <Logo
        logoClassName=""
      />
    </div>
  </nav>
  <div
    className="sidenav left false"
    id="navbarNavDropdown1"
  >
    <div
      className="container level-0"
    >
      <HamBurgerMenu
        defaultLeftNav={null}
        handleLeftMenu={[Function]}
        leftNavItems={
          Array [
            Object {
              "iconClass": "icon-home",
              "label": "home",
              "url": "home",
            },
            Object {
              "iconClass": "icon-promotions",
              "label": "inventory management",
              "url": "inventory-management",
            },
          ]
        }
      />
    </div>
    <div
      className="container level-1 false"
    >
      <HamBurgerSubMenu
        defaultSubNav={null}
        subMenuItems={Array []}
      />
    </div>
  </div>
</Fragment>
`;

exports[`Header Component render component without otherLinks 1`] = `
<Fragment>
  <nav
    className="navbar navbar-expand-md navbar-dark bg-dark fixed-top logged-in"
  >
    <div
      className="container"
    >
      <Logo
        logoClassName=""
      />
    </div>
  </nav>
  <div
    className="sidenav left false"
    id="navbarNavDropdown1"
  >
    <div
      className="container level-0"
    >
      <HamBurgerMenu
        defaultLeftNav={null}
        handleLeftMenu={[Function]}
        leftNavItems={
          Array [
            Object {
              "iconClass": "icon-home",
              "label": "home",
              "url": "home",
            },
            Object {
              "iconClass": "icon-promotions",
              "label": "inventory management",
              "url": "inventory-management",
            },
          ]
        }
      />
    </div>
    <div
      className="container level-1 false"
    >
      <HamBurgerSubMenu
        defaultSubNav={null}
        subMenuItems={Array []}
      />
    </div>
  </div>
</Fragment>
`;

exports[`Header Component render component without printer 1`] = `
<Fragment>
  <nav
    className="navbar navbar-expand-md navbar-dark bg-dark fixed-top logged-in"
  >
    <div
      className="container"
    >
      <Logo
        logoClassName=""
      />
    </div>
  </nav>
  <div
    className="sidenav left false"
    id="navbarNavDropdown1"
  >
    <div
      className="container level-0"
    >
      <HamBurgerMenu
        defaultLeftNav={null}
        handleLeftMenu={[Function]}
        leftNavItems={
          Array [
            Object {
              "iconClass": "icon-home",
              "label": "home",
              "url": "home",
            },
            Object {
              "iconClass": "icon-promotions",
              "label": "inventory management",
              "url": "inventory-management",
            },
          ]
        }
      />
    </div>
    <div
      className="container level-1 false"
    >
      <HamBurgerSubMenu
        defaultSubNav={null}
        subMenuItems={Array []}
      />
    </div>
  </div>
</Fragment>
`;

exports[`Header Component renders defaultProps correctly 1`] = `
<Fragment>
  <nav
    className="navbar navbar-expand-md navbar-dark bg-dark fixed-top logged-in"
  >
    <div
      className="container"
    >
      <Logo
        logoClassName=""
      />
    </div>
  </nav>
  <div
    className="sidenav left false"
    id="navbarNavDropdown1"
  >
    <div
      className="container level-0"
    >
      <HamBurgerMenu
        defaultLeftNav={null}
        handleLeftMenu={[Function]}
        leftNavItems={
          Array [
            Object {
              "iconClass": "icon-home",
              "label": "home",
              "url": "home",
            },
            Object {
              "iconClass": "icon-promotions",
              "label": "inventory management",
              "url": "inventory-management",
            },
          ]
        }
      />
    </div>
    <div
      className="container level-1 false"
    >
      <HamBurgerSubMenu
        defaultSubNav={null}
        subMenuItems={Array []}
      />
    </div>
  </div>
</Fragment>
`;

exports[`Header Component should open leftNavToggle onClick 1`] = `
<Fragment>
  <nav
    className="navbar navbar-expand-md navbar-dark bg-dark fixed-top logged-in"
  >
    <div
      className="container"
    >
      <Logo
        logoClassName=""
      />
      <button
        className="btn btn-sm btn-outline-primary"
        type="submit"
      >
        LOGIN
      </button>
    </div>
  </nav>
  <div
    className="sidenav left false"
    id="navbarNavDropdown1"
  >
    <div
      className="container level-0"
    >
      <HamBurgerMenu
        defaultLeftNav={null}
        handleLeftMenu={[Function]}
        leftNavItems={
          Array [
            Object {
              "iconClass": "icon-home",
              "label": "home",
              "url": "home",
            },
            Object {
              "iconClass": "icon-promotions",
              "label": "inventory management",
              "url": "inventory-management",
            },
          ]
        }
      />
    </div>
    <div
      className="container level-1 false"
    >
      <HamBurgerSubMenu
        defaultSubNav={null}
        subMenuItems={Array []}
      />
    </div>
  </div>
</Fragment>
`;

exports[`Header Component should render loggedOut header 1`] = `
<Fragment>
  <nav
    className="navbar navbar-expand-md navbar-dark bg-dark fixed-top logged-in"
  >
    <div
      className="container"
    >
      <Logo
        logoClassName=""
      />
      <button
        className="btn btn-sm btn-outline-primary"
        type="submit"
      >
        LOGIN
      </button>
    </div>
  </nav>
  <div
    className="sidenav left false"
    id="navbarNavDropdown1"
  >
    <div
      className="container level-0"
    >
      <HamBurgerMenu
        defaultLeftNav={null}
        handleLeftMenu={[Function]}
        leftNavItems={
          Array [
            Object {
              "iconClass": "icon-home",
              "label": "home",
              "url": "home",
            },
            Object {
              "iconClass": "icon-promotions",
              "label": "inventory management",
              "url": "inventory-management",
            },
          ]
        }
      />
    </div>
    <div
      className="container level-1 false"
    >
      <HamBurgerSubMenu
        defaultSubNav={null}
        subMenuItems={Array []}
      />
    </div>
  </div>
</Fragment>
`;

exports[`Header Component should render storeDetails from props 1`] = `
<Fragment>
  <nav
    className="navbar navbar-expand-md navbar-dark bg-dark fixed-top logged-in"
  >
    <div
      className="container"
    >
      <Logo
        logoClassName=""
      />
    </div>
  </nav>
  <div
    className="sidenav left false"
    id="navbarNavDropdown1"
  >
    <div
      className="container level-0"
    >
      <HamBurgerMenu
        defaultLeftNav={null}
        handleLeftMenu={[Function]}
        leftNavItems={
          Array [
            Object {
              "iconClass": "icon-home",
              "label": "home",
              "url": "home",
            },
            Object {
              "iconClass": "icon-promotions",
              "label": "inventory management",
              "url": "inventory-management",
            },
          ]
        }
      />
    </div>
    <div
      className="container level-1 false"
    >
      <HamBurgerSubMenu
        defaultSubNav={null}
        subMenuItems={Array []}
      />
    </div>
  </div>
</Fragment>
`;
